### PR TITLE
Highlight function defs

### PR DESCRIPTION
**Pull request by AndreasDahl**
Somewhat relates to the first point of #30. The difference being that all bindings are highlighted. not only functions. Let me know what you think.

NB!: this PR includes #31

### CODE REVIEW

**Comment by mroch** [comment]
i merged #31 before looking at this one, so I resolved the conflicts here

File: spec/ocaml_let_binding.spec [hunk 1]
```diff
@@ -1,37 +1,45 @@
 // SYNTAX TEST "source.ocaml"
-let x = 1 in 2
-//<- keyword.other.let_binding.ocaml
-//    ^ punctuation.separator.assignment.ocaml
-//        ^^ keyword.other.in.ocaml
-//^^^^^^^^^^ meta.let_binding.ocaml
-//          ^^ !meta.let_binding.ocaml
+let x a = 1 in 2
```
Comment (mroch) [inline spec/ocaml_let_binding.spec]: should `a` have a scope? `variable.parameter.ocaml`, perhaps (per http://manual.macromates.com/en/language_grammars#naming_conventions)

File: grammars/ocaml.cson [hunk 2]
```diff
@@ -27,16 +27,27 @@
     'name': 'meta.let_binding.ocaml'
     'patterns': [
       {
-        'comment': 'Matches everything between the `let|and` and the `=`'
```
Comment (mroch) [inline grammars/ocaml.cson]: mind adding a comment? this regex is crazytown

Comment (AndreasDahl) [inline grammars/ocaml.cson]: Thought about the regex some more and wrote a comment about it and simplified it. In the progress of simplifying it i also spotted a unhandled case
```
let rec ( a ) b c =
```
was not highlighted properly.

**Pull request merged**